disnake/interactions/base.py

```diff
@@ -160,60 +163,55 @@ def __init__(self, *, data: InteractionPayload, state: ConnectionState):
-        self._from_data(data)
-        self.bot: Optional[Bot] = None
 
-    def _from_data(self, data: InteractionPayload):
         self.id: int = int(data["id"])
         self.type: InteractionType = try_enum(InteractionType, data["type"])
         self.token: str = data["token"]
         self.version: int = data["version"]
-        self.channel_id: Optional[int] = utils._get_as_snowflake(data, "channel_id")
-        self.guild_id: Optional[int] = utils._get_as_snowflake(data, "guild_id")
         self.application_id: int = int(data["application_id"])
+
+        self.channel_id: int = int(data["channel_id"])
+        self.guild_id: Optional[int] = utils._get_as_snowflake(data, "guild_id")
         self.locale: str = data["locale"]
         self.guild_locale: Optional[str] = data.get("guild_locale")
-        # think about the user's experience
-        self.author: Union[User, Member] = None  # type: ignore
+        # one of user and member will always exist
+        self.author: Union[User, Member] = MISSING
         self._permissions: int = 0
 
-        # TODO: there's a potential data loss here
-        if self.guild_id:
-            guild = self.guild or Object(id=self.guild_id)
-            try:
-                member = data["member"]  # type: ignore
-            except KeyError:
-                pass
-            else:
-                self.author = (
-                    isinstance(guild, Guild)
-                    and guild.get_member(int(member["user"]["id"]))  # type: ignore
-                    or Member(state=self._state, guild=guild, data=member)  # type: ignore
-                )
-                self._permissions = int(member.get("permissions", 0))
-        else:
-            try:
-                self.author = User(state=self._state, data=data["user"])
-            except KeyError:
-                pass
+        if self.guild_id and (member := data.get("member")):
+            guild: Guild = self.guild or Object(id=self.guild_id)  # type: ignore
+            self.author = (
+                isinstance(guild, Guild)
+                and guild.get_member(int(member["user"]["id"]))
+                or Member(state=self._state, guild=guild, data=member)
+            )
+            self._permissions = int(member.get("permissions", 0))
+        elif user := data.get("user"):
+            self.author = self._state.store_user(user)
 
     @property
-    def created_at(self) -> datetime:
-        return utils.snowflake_time(self.id)
+    def bot(self) -> AnyBot:
+        """:class:`~disnake.ext.commands.Bot`: The bot handling the interaction.
+
+        Only applicable when used with :class:`~disnake.ext.commands.Bot`.
+        This is an alias for :attr:`.client`.
+        """
+        return self.client  # type: ignore
 
     @property
-    def client(self) -> Optional[Bot]:
-        return self.bot
+    def created_at(self) -> datetime:
+        return utils.snowflake_time(self.id)
```

Comment:
Oh by the way I heard that this datetime is usually offset by several seconds (about 4), not sure why exactly. There was an additional attribute called `received_at` in dislash that was basically a datetime object representing the time when the interaction was received. What do you think about adding it in this PR? We can discuss this in discord